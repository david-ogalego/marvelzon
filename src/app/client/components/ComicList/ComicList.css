.list {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    display: -ms-flexbox;
    -ms-flex-direction: row;
    -ms-flex-wrap: wrap;
    margin: 80px 25px 25px 25px;
}

.containerLoadMore {
    width: 100%;
}

button {
    background: none;
    border: none;
    outline: none;
}

.loadMore {
    height: 40px;
    cursor: pointer;
}

.spinner {
    width: 50px;
    height: 50px;
    background-color: #0ec3ca;
    position: fixed;
    top: 50%;
    left: 48%;
    border-radius: 100%;  
    -webkit-animation: sk-scaleout 1.0s infinite ease-in-out;
    animation: sk-scaleout 1.0s infinite ease-in-out;
}
  
@-webkit-keyframes sk-scaleout {
    0% {
        -webkit-transform: scale(0)
    }
    100% {
        -webkit-transform: scale(1.0);
        opacity: 0;
    }
}
  
@keyframes sk-scaleout {
    0% { 
        -webkit-transform: scale(0);
        transform: scale(0);
    } 100% {
        -webkit-transform: scale(1.0);
        transform: scale(1.0);
        opacity: 0;
    }
}

@media only screen and (min-device-width : 320px) and (max-device-width : 640px) {
    .loadMore {
        margin-left: 45%;
    }
}