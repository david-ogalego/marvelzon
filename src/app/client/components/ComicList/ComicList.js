import React from 'react';
import { ComicItem } from '../';
import { getImageFromComic, getDateFromComic, getPriceFromComic} from '../../helpers/comic';
import loadMoreIcon from '../../images/loadMore.png';
import stylesComicList from './ComicList.css';

const ComicList = function(props) {
    return (
        <div>
            {props.showLoading && <div className={stylesComicList.spinner}></div>}
            {!!props.comicsList.length &&
                <div className={stylesComicList.list}>
                    {props.comicsList.map((comic, index) => 
                        <ComicItem
                            key={index}
                            title={comic.title}
                            id={comic.id}
                            date={getDateFromComic(comic)}
                            price={getPriceFromComic(comic)}
                            image={getImageFromComic(comic)} />
                    )}
                    <div className={stylesComicList.containerLoadMore} onClick={() => {
                        if (props.onLoadMore) {
                            props.onLoadMore();
                        }
                    }}>
                        <img className={stylesComicList.loadMore} alt="Load more" src={loadMoreIcon}/>
                    </div>
                </div>
            }
        </div>
    )
    
};

export default ComicList;