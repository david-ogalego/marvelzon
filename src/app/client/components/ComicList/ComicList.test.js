import React from 'react';
import renderer from 'react-test-renderer';
import ComicList from './ComicList';
import { MemoryRouter } from 'react-router-dom';

const comicList = [
    {
        id: 0,
        title: 'X-Men',
        dates: [
            {
                date: '2029-12-31T00:00:00-0500',
                type: 'onsaleDate'
            }
        ],
        prices: [
            {
                price: 3.99,
                type: 'printPrice'
            }
        ],
        thumbnail: {
            path: "http://i.annihil.us/u/prod/marvel/i/mg/9/90/4bc6353e5fc56",
            extension: "jpg"
        },
        pageCount: 10,
        description: 'Lorem Ipsum ha sido el texto de relleno estándar de las industrias desde el año 1500, cuando un impresor (N. del T. persona que se dedica a la imprenta) desconocido usó una galería de textos',
        url: 'http://marvel.com/comics/issue/2088/official_handbook_of_the_marvel_universe_2004_14_fantastic_four/fantastic_four?utm_campaign=apiRef&utm_source=d86beaee5f52cf5b1205630a7e35b24b'
    },
    {
        id: 1,
        title: 'X-Men-2',
        dates: [
            {
                date: '2018-12-31T00:00:00-0500',
                type: 'onsaleDate'
            }
        ],
        prices: [
            {
                price: 100,
                type: 'printPrice'
            }
        ],
        thumbnail: {
            path: "http://i.annihil.us/u/prod/marvel/i/mg/9/90/4bcsjnfsnf",
            extension: "jpg"
        },
        pageCount: 100,
        description: 'El punto de usar Lorem Ipsum es que tiene una distribución más o menos normal de las letras, al contrario de usar textos como por ejemplo "Contenido aquí, contenido aquí".',
        url: 'http://marvel.com/comics/issue/2088/official_handbook_of_the_marvel_universe_2004_14_fantastic_four/fantastic_four?utm_campaign=apiRef&utm_source=d86beaee5f52cf5b1205630a7e35b24b'
    }
];

test('Snapshot ComicList rendered loading', () => {
    const treeComicList = renderer.create(
            <MemoryRouter>
                <ComicList showLoading={true} comicsList={[]} />
            </MemoryRouter>
        ).toJSON();
    expect(treeComicList).toMatchSnapshot();
});

test('Snapshot ComicList rendered with comic list', () => {
    const treeComicList = renderer.create(
        <MemoryRouter>
            <ComicList showLoading={true} comicsList={comicList} />
        </MemoryRouter>
        ).toJSON();
    expect(treeComicList).toMatchSnapshot();
});
