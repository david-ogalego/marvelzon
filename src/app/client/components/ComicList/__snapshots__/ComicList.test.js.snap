// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ComicList rendered loading 1`] = `
<div>
  <div
    className="spinner"
  />
</div>
`;

exports[`Snapshot ComicList rendered with comic list 1`] = `
<div>
  <div
    className="spinner"
  />
  <div
    className="list"
  >
    <a
      className="item"
      href="/detail/0"
      onClick={[Function]}
    >
      <div
        className="containerImage"
      >
        <img
          alt="X-Men"
          className="image"
          src="https://i.annihil.us/u/prod/marvel/i/mg/9/90/4bc6353e5fc56/portrait_incredible.jpg"
        />
        <div
          className="containerDetails"
        >
          <div
            className="detail titleDetail"
          >
            X-Men
          </div>
          <div
            className="detail"
          >
            2029-12-31
          </div>
          <div
            className="detail"
          >
            Price: 
            3.99
          </div>
        </div>
      </div>
    </a>
    <a
      className="item"
      href="/detail/1"
      onClick={[Function]}
    >
      <div
        className="containerImage"
      >
        <img
          alt="X-Men-2"
          className="image"
          src="https://i.annihil.us/u/prod/marvel/i/mg/9/90/4bcsjnfsnf/portrait_incredible.jpg"
        />
        <div
          className="containerDetails"
        >
          <div
            className="detail titleDetail"
          >
            X-Men-2
          </div>
          <div
            className="detail"
          >
            2018-12-31
          </div>
          <div
            className="detail"
          >
            Price: 
            100
          </div>
        </div>
      </div>
    </a>
    <div
      className="containerLoadMore"
      onClick={[Function]}
    >
      <img
        alt="Load more"
        className="loadMore"
        src="test-file-stub"
      />
    </div>
  </div>
</div>
`;
